hack/run-hive-locally.sh

```diff
@@ -0,0 +1,365 @@
+    while read -r env_value; do
+        env="${env_value%=*}"
+
+        # Skip environment variables that are already resolved
+        if [ ${resolved_envs["$env"]+_} ]; then
+            hl_echo "Environment variable $env already resolved, skipping"
+            continue
+        fi
+
+        # Skip irrelevant environment variables
+        if [[ $env = "CLI_CACHE_DIR" ]] || [[ $env = "HIVE_SKIP_LEADER_ELECTION" ]]; then
+            hl_echo "Environment variable $env is irrelevant, skipping"
+            continue
+        fi
+
+        # Resolve the environment variable's value
+        value="${env_value#*=}"
+        case "$value" in
+        "")
+            hl_echo "Environment variable $env does not have a .value, skipping"
+            continue
+            ;;
+        /*)
+            resolved_value="./$artifacts_folder/$(basename "$value")-${hc_spec_generation}.txt"
+            # Skip if the file does not exist
+            if ! oc exec "$type_name" -n "$ns" -- /bin/bash -c "cat $value" > "$resolved_value"; then
+                rm "$resolved_value"
+                continue
+            fi
+            hl_echo "File content saved to $resolved_value"
+            ;;
+        *)
+            resolved_value="$value"
+            ;;
+        esac
+
+        resolved_envs["$env"]="$env=$resolved_value"
+        hl_echo "Environment variable $env=$resolved_value"
+    done < <(oc get "$type_name" -n "$ns" -o jsonpath="$jsonpath_query")
+}
+
+function get_envs_from {
+    local type_name="$1"
+    local ns="$2"
+    # Arrange output into the following format:
+    # ConfigMap1
+    # ConfigMap2
+    # ...
+    local jsonpath_query='{range .spec.template.spec.containers[0].envFrom[*]}{@.configMapRef.name}{"\n"}{end}'
+    # Arrange output into the following format:
+    # key1=value1
+    # key2=value2
+    # ...
+    local jq_query='to_entries[] | "\(.key)=\(.value)"'
+    local cm
+    local env_value
+    local env
+    local value
+    hl_echo "Getting envFroms from $type_name in ns/$ns"
+
+    while read -r cm; do
+        # Skip ConfigMaps that are already resolved
+        if [ ${resolved_envFrom_cms["$cm"]+_} ]; then
+            hl_echo "cm/$cm already resolved, skipping"
+            continue
+        fi
+
+        # Extract environment variables from ConfigMap
+        hl_echo "Extracting environment variables from cm/$cm in ns/$ns"
+        while read -r env_value; do
+            env="${env_value%=*}"
+            value="${env_value#*=}"
+            resolved_envs["$env"]="$env=$value"
+            hl_echo "Environment variable $env = $value"
+        done < <(oc get "cm/$cm" -n "$ns" -o jsonpath="{.data}" | jq -r "$jq_query")
+
+        # Mark ConfigMap as resolved
+        resolved_envFrom_cms["$cm"]=""
+
+        # Also save ConfigMap.data for future reference
+        oc get "cm/$cm" -n "$ns" -o jsonpath="{.data}" > "./$artifacts_folder/$cm-${hc_spec_generation}.txt"
```

Comment:
Similar comment about using the generation as the prefix.

Also it might not be a terrible idea to put the actual string `-cm-` in the file name to make it easy to spot that it came from a configmap?

Reply:
Fixed. Now it looks like `0-cm-hive-controllers-config.txt` for example.